src_c/math.c

```diff
@@ -1287,6 +1293,89 @@ vector_scale_to_length(pgVector *self, PyObject *length)
     Py_RETURN_NONE;
 }
 
+static int
+_vector_move_towards_helper(Py_ssize_t dim, double *origin_coords, double *target_coords, double distance)
+{
+    Py_ssize_t i;
+    double delta[VECTOR_MAX_SIZE];
+    double dist;
+
+    if (distance < 0) {
+        PyErr_SetString(PyExc_TypeError, 
+            "Expected distance to be greater or equal to 0");
+        return 0;
+    }
+
+    for (i = 0; i < dim; ++i)
+    {
+        delta[i] = origin_coords[i] - target_coords[i];
+    }
+
+    /* Get magnitude of Vector */
+    dist = sqrt(_scalar_product(delta, delta, dim));
+
+    if (dist <= distance || dist == 0)
+    {
+        /* Return target Vector */
+        for (i = 0; i < dim; ++i)
+            origin_coords[i] = target_coords[i];
+        return 1;
+    }
+
+    for (i = 0; i < dim; ++i)
+        origin_coords[i] = origin_coords[i] - delta[i] / dist * distance;
+
+    return 1;
+}
+
+static PyObject *
+vector_move_towards(pgVector *self, PyObject *args)
+{
+    Py_ssize_t i;
+    pgVector *target;
+    double distance;
+    pgVector *ret;
+    ret = (pgVector *)pgVector_NEW(self->dim);
+    if (ret == NULL) {
+        return NULL;
+    }
+    for (i = 0; i < self->dim; ++i)
+        ret->coords[i] = self->coords[i];
+
+    if (!PyArg_ParseTuple(args, "Od:move_towards", &target, &distance)) {
+        return NULL;
```

Comment:
hmmm this can potentially segfault if the first argument passed is not a Vector object, see other functions for how the conversion is done. Also this arg handling bit should come at the top, before the `pgVector_NEW` call, because it should error quickly when args are not right.